api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -191,6 +191,11 @@ HttpClient getHttpClient() {
         .build();
   }
 
+  void clearSecrets() {
```

Comment:
A corollary of this change is that secrets will be kept in the config object until the client is closed. I think we can live with that.

Reply:
However, that changes the semantics of an _immutable_ object.
WDYT of removing the ability to clear entirely, but make the accessors in `Secret` package-private?

Reply:
Actually `Secret` was introduced precisely to isolate all the mutable stuff in a class that is not `@Immutable`. 

Let me try your suggestion.

Reply:
I tend to think that clearing these secrets is not very useful because the client is long-lived. I believe JRE crypto method generally clear passwords immediately after a single use, but in our case we use them many times.

+1 to reducing visibility of those methods.